source/lzss.cpp

```diff
@@ -83,13 +83,18 @@ const uint8_t *find_best_match (const uint8_t *start,
 	const uint8_t *best_start = buffer;
 	size_t best_len           = 0;
 
+	const uint8_t *p = buffer;
+
 	// find nearest matching start byte
-	const uint8_t *p = rfind (start, buffer, *buffer);
-	while (p)
+	while ((p = rfind (start, p, *buffer)))
 	{
+		// confirm that this match is as good as the best match so far
+		if(memcmp(p, buffer, best_len))
```

Comment:
```cpp
if (std::memcmp (p, buffer, best_len) != 0)
```

Reply:
That does make it more explicit, I'll be sure to do that. Want me to make the above line `while ((p = rfind (start, p, *buffer)) != nullptr)` instead of `while ((p = rfind (start, p, *buffer)))` as well?

Reply:
(Or, I could use a for loop instead: `for (p = rfind (start, buffer, *buffer); p; p = rfind (start, p, *buffer))`)

Reply:
Nah the while loop condition is fine. I prefer the explicit memcmp != 0 because I consider 0=success to be inverted logic which is a common source of bugs.

Reply:
Done, thanks!